external/source/exploits/CVE-2022-34918/src/util.c

```diff
@@ -0,0 +1,154 @@
+#define _GNU_SOURCE
+#include <stdint.h>
+#include <stdlib.h>
+#include <sched.h>
+#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+#include "log.h"
+#include "util.h"
+
+struct kernel_info kernels[] = {
+	{ "5.15.0-39-generic #42-Ubuntu", 0x3dfa00, 0x3e04f0, 0x1e8b620 },
+	{ "5.15.0-40-generic #43-Ubuntu", 0x3dfa00, 0x3e04f0, 0x1e8b620 },
+	{ "5.15.0-25-generic #25-Ubuntu", 0x3dda20, 0x3de520, 0x1e8b3a0 }
+};
```

Comment:
As you're using the `kernel_info` struct as discussed on Slack, you can incorporate this into the `check` method.

Here's an example:

https://github.com/rapid7/metasploit-framework/blob/e46a71f59539cb17d7cbf3f32ca5980793dfd25a/modules/exploits/linux/local/netfilter_xtables_heap_oob_write_priv_esc.rb#L91-L98